controllers/glanceapi_controller.go

```diff
@@ -609,6 +612,31 @@ func (r *GlanceAPIReconciler) reconcileNormal(ctx context.Context, instance *gla
 	return ctrl.Result{}, nil
 }
 
+func (r *GlanceAPIReconciler) validateMons(ipList string) bool {
+	for _, ip := range strings.Split(ipList, ",") {
+		if net.ParseIP(strings.Trim(ip, " ")) == nil {
+			return false
+		}
+	}
+	return true
+}
+
+/*  The GlanceBackend is set to file by default. However, if Ceph parameters are
+provided means we want to use it as backend. This function checks the Ceph
+related parameters and returns the right backend.
+*/
+func (r *GlanceAPIReconciler) setGlanceBackend(instance *glancev1beta1.GlanceAPI) string {
+	// Validate parameters to enable CephBackend
+	if instance.Spec.CephBackend.CephClusterFSID != "" &&
+		r.validateMons(instance.Spec.CephBackend.CephClusterMonHosts) &&
+		instance.Spec.CephBackend.CephClientKey != "" {
+		return "rbd"
+	}
+	/* file is the default backend: if we don't have any ceph
+	parameter set backend to file */
+	return "file"
+}
```

Comment:
Ack and thanks for your feedback.
Given those are `utility` functions, I guess it's better moving them to `pkg/glance` and keep this section as slim as possible.
As you said, this is useful to make sure the controller has no clues about the underlying CR lifecyle and can be a good example to build other OSP operators prototypes.

Reply:
I guess this can be resolved, now those functions are moved and documented in `pkg/glance`, where only SetBackend is exposed.
In the future, as long as more backends are supported, we can think about adding a util.go, but that's pure code organization/refactoring that can be a follow up work.